src/esp/scene/SceneManager.h

```diff
@@ -30,6 +27,9 @@ class SceneManager {
   SceneGraph& getSceneGraph(int sceneID);
   const SceneGraph& getSceneGraph(int sceneID) const;
 
+  // Returns the count of registered scene graphs
+  inline int getSceneGraphCount() { return sceneGraphs_.size(); }
```

Comment:
Wow, implicit loss of percision here:
```suggestion
  inline size_t getSceneGraphCount() { return sceneGraphs_.size(); }
```